routes/v1.js

```diff
@@ -149,37 +228,483 @@ api.declare({
+  scopes:   [[
+    'aws-provisioner:all-stop',
+    'aws-provisioner:aws'
+  ]],
+  description: [
+    "WARNING: YOU ALMOST CERTAINLY DO NOT WANT TO USE THIS ",
+    "Shut down every single EC2 instance managed by this provisioner. ",
+    "This means every single last one.  You probably don't want to use ",
+    "this method, which is why it has an obnoxious name.  Don't even try ",
+    "to claim you didn't know what this method does!"
+  ].join('\n')
+}, function(req, res) {
+
+  var ctx = this;
+
+  debug('SOMEONE IS TURNING EVERYTHING OFF');
+
+  // Note that by telling the rouge killer
+  var p = ctx.awsManager.rougeKiller([]);
+
+  p = p.then(function() {
+    res.reply({
+      outcome: true,
+      message: 'You just turned absolutely everything off.  Feel the power!',
+    });
+  });
+
+  p = p.catch(function(err) {
+    console.error(err, err.stack);
+    res.status(503).json({
+      message: 'Could not shut down everything',
+    });
+  });
+
+  return p;
+
+});
+
+
+api.declare({
+  method:         'post',
+  route:          '/worker-type/:workerType/spawn',
+  name:           'spawnWorkerType',
+  deferAuth:      true,
+  scopes:         [[
+    'aws-provisioner:aws',
+    'aws-provisioner:spawn-worker-type:<workerType>'
+  ]],
+  input:          SCHEMA_PREFIX_CONST + 'spawn-instance-request.json#',
+  output:         SCHEMA_PREFIX_CONST + 'spawn-instance-response.json#',
+  title:          "Spawn an instance of a Worker Type",
+  description: [
+    'Spawn an instance of a workerType given a bid.  A bid is an ',
+    'object which has a number `price`, string `region` and string ',
+    '`instance type`.'
+  ].join('\n')
+}, function(req, res) {
+  var ctx         = this;
+  var input       = req.body;
+  var workerType  = req.params.workerType;
+  
+  if(!req.satisfies({
+    workerType:       workerType
+  })) {
+    return; // by default req.satisfies() sends a response on failure, so we're done
+  }
+
+  var p = ctx.WorkerType.load({workerType: workerType})
+    
+  p = p.then(function(worker) {
+    return ctx.awsManager.requestSpotInstance(worker, input);
+  });
+
+  p = p.then(function(sir) {
+    var reply = {
+      SpotInstanceRequest: sir,
+    };
+    return res.reply(reply);
+  })
+
+  p = p.catch(function(err) {
+    errorHandler(err, res, workerType);
+    return err;
+  });
+
+  return p;
+
+});
+
+
+api.declare({
+  method:         'get',
+  route:          '/worker-type/:workerType/stats',
+  name:           'workerTypeStats',
+  deferAuth:      true,
+  scopes:         [[
+    'aws-provisioner:get-worker-type:<workerType>'
+  ]],
+  output:         SCHEMA_PREFIX_CONST + 'worker-type-stats-response.json#',
+  title:          "Spawn an instance of a Worker Type",
+  description: [
+    'Spawn an instance of a workerType given a bid.  A bid is an ',
+    'object which has a number `price`, string `region` and string ',
+    '`instance type`.'
```

Comment:
declare experimental if you're not defining a more clear JSON schema... We probably don't critically need this now.